File: testdata/bench/rust/BenchmarkTest00088.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: text/plain content type prevents XSS
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("msg").unwrap_or(&String::new()).clone();
    HttpResponse::Ok().content_type("text/plain").body(param)
}

File: testdata/bench/cpp/BenchmarkTest00088.cpp
#include <sstream>
#include <iostream>
#include <cstdlib>

int main() {
    char *val = getenv("QUERY_STRING");
    std::ostringstream oss;
    oss << "<html><body><div class='msg'>" << (val ? val : "") << "</div></body></html>";
    std::cout << "Content-Type: text/html\r\n\r\n" << oss.str();
    return 0;
}
